base/osutils.jl

```diff
@@ -1,20 +1,26 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
-"""
-    OS_NAME
+let UNAME = ccall(:jl_get_UNAME, Any, ())
+    # change some `uname -s` values to "friendly" names
+    # for internal usage
```

Comment:
We're off in the weeds, but mainly I don't think `Sys.KERNEL` should be called that.

The benefit of WSL to end-users is being able to do things like `apt-get install openmpi-bin` and have it work without having to teach every single build system in the world what to do with a new `uname` value. It's disconnected from the win32 subsystem, but uses the same NT kernel - just not in a way that's exposed to the user, who never needs to worry about system calls. Supporting WSL from a Julia standpoint will just be a matter of running our Linux binaries there, unmodified, and reporting to Microsoft when things don't work.